x/pocketcore/keeper/proof.go

```diff
@@ -20,33 +20,127 @@ func BeginBlocker(ctx sdk.Ctx, _ abci.RequestBeginBlock, k Keeper) {
 	k.DeleteExpiredClaims(ctx)
 }
 
-// validate the zero knowledge range proof using the proof message and the claim message
-func (k Keeper) ValidateProof(ctx sdk.Ctx, claimMsg pc.MsgClaim, proofMsg pc.MsgProof) error {
-	// generate the needed pseudorandom claimMsg index
-	ctx.Logger().Info(fmt.Sprintf("Generate psuedorandom proof with %d, at session height of %d", claimMsg.TotalRelays, claimMsg.SessionBlockHeight))
-	reqProof, err := k.GetPseudorandomIndex(ctx, claimMsg.TotalRelays, claimMsg.SessionHeader)
+// auto sends a proof transaction for the claim
+func (k Keeper) SendProofTx(ctx sdk.Ctx, n client.Client, keybase keys.Keybase, proofTx func(cliCtx util.CLIContext, txBuilder auth.TxBuilder, branches [2]pc.MerkleProof, leafNode, cousin pc.Proof) (*sdk.TxResponse, error)) {
+	kp, err := keybase.GetCoinbase()
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("an error occured retrieving the coinbase for the ProofTX:\n%v", err))
+		return
+	}
+	// get the self address
+	addr := kp.GetAddress()
+	// get all mature (waiting period has passed) claims for your address
+	claims, err := k.GetMatureClaims(ctx, addr)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("an error occured getting the mature claims in the ProofTX:\n%v", err))
+		return
+	}
+	// for every claim of the mature set
+	for _, claim := range claims {
+		// if the claim is found to be verified in the world state, you can delete it from the cache and not send again
+		if _, found := k.GetReceipt(ctx, addr, claim.SessionHeader, claim.EvidenceType); found {
+			// remove from the local cache
+			pc.GetEvidenceMap().DeleteEvidence(claim.SessionHeader, claim.EvidenceType)
+			continue
+		}
+		// check to see if evidence is stored in cache
+		evidence, found := pc.GetEvidenceMap().GetEvidence(claim.SessionHeader, claim.EvidenceType)
+		if !found || evidence.Proofs == nil || len(evidence.Proofs) == 0 {
+			ctx.Logger().Info(fmt.Sprintf("the evidence object for evidence is not found, ignoring pending claim"))
```

Comment:
This log would be more useful if we somehow identified the ignored pending claim.

Reply:
Will fix